src/Configuration/SshParametersInterface.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\DbExtractorCommon\Configuration;
+
+interface SshParametersInterface
```

Comment:
Nemyslím si, že ten interface dává smysl. Ty věci v tom interface jsou nutné podmínky toho, abys měl to SSH. Nemůžeš si udělat žádnou jinou implementaci, která bude mít něco navíc. Naopak bych si myslel, že SshParameters může být final. Prostě nic jiného nepotřebuješ. Ale může být extra třída, která ti ty SshParameters udělá a ta by mohla mít interface. `interface SshParametersFactory { fromDbConfig($db): SshParameters }`. Bude na každém tvůrci extractoru, aby si naimplementoval, jak z DB configu udělá ten pevný a neměnný objekt.

Reply:
mmch myslím, že by to mělo dostávat raw config, ne ten DbParameters objekt, protože nevíme, co všechno z něj bude potřeba. Třeba nějaká DB má dvě hesla, nebo heslo a klíč a podle nich se nastavuje ten tunel. A když bysme typehintovali ten objekt, tak to zamkneme jen na metody toho interfacu.